### PR TITLE
docs(topology): Add props tables to topology docs

### PR DESCRIPTION
**Pull request by jeff-phillips-18**
**What**:
Closes https://github.com/patternfly/patternfly-extension-issues/issues/36

### CODE REVIEW

File: packages/react-topology/src/components/edges/DefaultEdge.tsx [hunk 1]
```diff
@@ -20,28 +14,58 @@ import DefaultConnectorTag from './DefaultConnectorTag';
...
 
-type BaseEdgeProps = {
+interface DefaultEdgeProps {
+  /** Additional content added to the edge */
   children?: React.ReactNode;
+  /** Additional classes added to the edge */
+  className?: string;
+  /** The graph edge element to represent */
   element: Edge;
+  /** Is the user dragging the edge */
```
Comment (jenny-s51) [inline packages/react-topology/src/components/edges/DefaultEdge.tsx:24]:
```suggestion
  /** Flag indicating if the user is dragging the edge */
```

File: packages/react-topology/src/components/edges/DefaultEdge.tsx [hunk 2]
```diff
@@ -20,28 +14,58 @@ import DefaultConnectorTag from './DefaultConnectorTag';
...
->;
+  /** Function to call for showing a remove indicator on the edge. Part of WithRemoveConnectorProps  */
+  onShowRemoveConnector?: () => void;
+  /** Function to call for removing the remove indicator on the edge. Part of WithRemoveConnectorProps  */
+  onHideRemoveConnector?: () => void;
+  /** ref to use to start the drag of the start of the edge. Part of WithSourceDragProps */
+  sourceDragRef?: ConnectDragSource;
+  /** ref to use to start the drag of the end of the edge. Part of WithTargetDragProps */
+  targetDragRef?: ConnectDragSource;
+  /** Is the element selected. Part of WithSelectionProps */
```
Comment (jenny-s51) [inline packages/react-topology/src/components/edges/DefaultEdge.tsx:58]:
```suggestion
  /** Flag indicating if the element is selected. Part of WithSelectionProps */
```

File: packages/react-topology/src/components/nodes/DefaultNode.tsx [hunk 3]
```diff
@@ -36,49 +29,94 @@ const getStatusIcon = (status: NodeStatus) => {
...
   className?: string;
+  /** The graph node element to represent */
   element: Node;
+  /** Does the node accept drop operations */
   droppable?: boolean;
+  /** Is the user hovering on the node */
   hover?: boolean;
+  /** Can the current drag operation be dropped on the node */
   canDrop?: boolean;
+  /** Is the user dragging the node */
```
Comment (jenny-s51) [inline packages/react-topology/src/components/nodes/DefaultNode.tsx:45]:
```suggestion
  /** Flag indicating if the user is dragging the node */
```

File: packages/react-topology/src/components/nodes/DefaultNode.tsx [hunk 4]
```diff
@@ -36,49 +29,94 @@ const getStatusIcon = (status: NodeStatus) => {
...
   badgeClassName?: string;
+  /** Location of the badge relative to the label's text, inner or below */
   badgeLocation?: BadgeLocation;
-  attachments?: React.ReactNode; // ie. decorators
-  nodeStatus?: NodeStatus; // Defaults to element.getNodeStatus()
+  /** Additional items to add to the node, typically decorators */
+  attachments?: React.ReactNode;
+  /** Status of the node, Defaults to element.getNodeStatus() */
+  nodeStatus?: NodeStatus;
+  /** Should the node's background indicate status */
```
Comment (jenny-s51) [inline packages/react-topology/src/components/nodes/DefaultNode.tsx:89]:
```suggestion
  /** Flag indicating whether the node's background color should match the status color */
```

File: packages/react-topology/src/components/nodes/DefaultNode.tsx [hunk 5]
```diff
@@ -36,49 +29,94 @@ const getStatusIcon = (status: NodeStatus) => {
...
   canDrop?: boolean;
+  /** Is the user dragging the node */
   dragging?: boolean;
+  /** Is the user dragging an edge connected to the node */
   edgeDragging?: boolean;
+  /** Is the node the current drop target */
   dropTarget?: boolean;
-  scaleNode?: boolean; // Whether or not to scale the node, best on hover of node at lowest scale level
-  label?: string; // Defaults to element.getLabel()
+  /** Should the node be scaled, best on hover of node at lowest scale level */
```
Comment (jenny-s51) [inline packages/react-topology/src/components/nodes/DefaultNode.tsx:51]:
```suggestion
  /** Flag indicating the node should be scaled, best on hover of node at lowest scale level */
```

File: packages/react-topology/src/components/nodes/DefaultNode.tsx [hunk 6]
```diff
@@ -36,49 +29,94 @@ const getStatusIcon = (status: NodeStatus) => {
...
+  scaleNode?: boolean;
+  /** Label for the node, Defaults to element.getLabel() */
+  label?: string;
+  /** Secondary label for the node */
   secondaryLabel?: string;
-  showLabel?: boolean; // Defaults to true
+  /** Should the label be shown */
```
Comment (jenny-s51) [inline packages/react-topology/src/components/nodes/DefaultNode.tsx:57]:
```suggestion
  /** Flag to show the label */
```

File: packages/react-topology/src/components/nodes/DefaultNode.tsx [hunk 7]
```diff
@@ -36,49 +29,94 @@ const getStatusIcon = (status: NodeStatus) => {
...
+  showLabel?: boolean;
+  /** Additional classes to add to the label */
   labelClassName?: string;
-  scaleLabel?: boolean; // Whether or not to scale the label, best at lower scale levels
-  labelPosition?: LabelPosition; // Defaults to element.getLabelPosition()
-  truncateLength?: number; // Defaults to 13
-  labelIconClass?: string; // Icon to show in label
+  /** Should the label be scaled, best on hover of node at lowest scale level */
```
Comment (jenny-s51) [inline packages/react-topology/src/components/nodes/DefaultNode.tsx:61]:
```suggestion
  /** Flag to scale the label, best on hover of node at lowest scale level */
```

File: packages/react-topology/src/components/nodes/DefaultNode.tsx [hunk 8]
```diff
@@ -36,49 +29,94 @@ const getStatusIcon = (status: NodeStatus) => {
...
   badgeLocation?: BadgeLocation;
-  attachments?: React.ReactNode; // ie. decorators
-  nodeStatus?: NodeStatus; // Defaults to element.getNodeStatus()
+  /** Additional items to add to the node, typically decorators */
+  attachments?: React.ReactNode;
+  /** Status of the node, Defaults to element.getNodeStatus() */
+  nodeStatus?: NodeStatus;
+  /** Should the node's background indicate status */
   showStatusBackground?: boolean;
+  /** Should the status decorator be displayed for the node */
```
Comment (jenny-s51) [inline packages/react-topology/src/components/nodes/DefaultNode.tsx:91]:
```suggestion
  /** Flag which displays the status decorator for the node */
```

**Review by jenny-s51 — CHANGES_REQUESTED** [review]
Such an exciting update!! 

Added some suggestions to align the prop descriptions - looks like the PF convention is to use "Flag" whenever describing boolean props. 

Otherwise prop tables are looking awesome @jeff-phillips-18 🎉

**Review by jenny-s51 — COMMENTED** [review]
Just needs a few more small edits

**Review by jenny-s51 — APPROVED** [review]
👍

File: packages/react-topology/src/components/VisualizationProvider.tsx [hunk 9]
```diff
@@ -3,12 +3,17 @@ import ControllerContext from '../utils/ControllerContext';
 import { Controller } from '../types';
 import { Visualization } from '../Visualization';
 
-interface VisualizationSurfaceProps {
+interface VisualizationProviderProps {
+  /** The graph Controller to store in context */
```
Comment (nicolethoen) [inline packages/react-topology/src/components/VisualizationProvider.tsx:7]: lowercase controller.

File: packages/react-topology/src/components/VisualizationSurface.tsx [hunk 10]
```diff
@@ -16,7 +16,9 @@ import useVisualizationController from '../hooks/useVisualizationController';
 import '@patternfly/react-styles/css/components/Topology/topology-components.css';
 
 interface VisualizationSurfaceProps {
+  /** state to be passed to the controller */
```
Comment (nicolethoen) [inline packages/react-topology/src/components/VisualizationSurface.tsx:19]: Capitalize State

File: packages/react-topology/src/components/nodes/DefaultNode.tsx [hunk 11]
```diff
@@ -36,49 +29,94 @@ const getStatusIcon = (status: NodeStatus) => {
...
   dragging?: boolean;
+  /** Flag if the user is dragging an edge connected to the node */
   edgeDragging?: boolean;
+  /** Flag if the node is the current drop target */
   dropTarget?: boolean;
-  scaleNode?: boolean; // Whether or not to scale the node, best on hover of node at lowest scale level
-  label?: string; // Defaults to element.getLabel()
+  /** Flag indicating the node should be scaled, best on hover of the node at lowest scale level */
+  scaleNode?: boolean;
+  /** Label for the node, Defaults to element.getLabel() */
```
Comment (nicolethoen) [inline packages/react-topology/src/components/nodes/DefaultNode.tsx:53]: update comma to period

File: packages/react-topology/src/components/edges/DefaultEdge.tsx [hunk 12]
```diff
@@ -20,28 +14,58 @@ import DefaultConnectorTag from './DefaultConnectorTag';
...
+  /** The status to indicate on the tag */
   tagStatus?: NodeStatus;
-} & Partial<
-  WithRemoveConnectorProps & WithSourceDragProps & WithTargetDragProps & WithSelectionProps & WithContextMenuProps
->;
+  /** Function to call for showing a remove indicator on the edge. Part of WithRemoveConnectorProps  */
+  onShowRemoveConnector?: () => void;
+  /** Function to call for removing the remove indicator on the edge. Part of WithRemoveConnectorProps  */
+  onHideRemoveConnector?: () => void;
+  /** ref to use to start the drag of the start of the edge. Part of WithSourceDragProps */
```
Comment (nicolethoen) [inline packages/react-topology/src/components/edges/DefaultEdge.tsx:54]: capitalize Ref here and in next comment

**Review by kmcfaul — APPROVED** [review]

**Review by jenny-s51 — APPROVED** [review]

**Pull request merged**